### ISSUE TITLE
Add support for Navi Controller

### ISSUE THREAD

**Issue by HipsterSloth** [cboulay/PSMoveService#42]
The framework for this controller is mostly done. The controller pairing and polling doesn't work yet. And there are probably a bunch of places in the code still hard coded to assume a psmove.

**Comment by andregm3** [comment]
![j9ijv7cltqvcoklsd6ctbsx4etqsslpyx0ejwxq84-u-555x368](https://cloud.githubusercontent.com/assets/4154732/16901229/96ef61d0-4c13-11e6-9c11-d11cd49c516d.jpg)
![dualplay_peripheral](https://cloud.githubusercontent.com/assets/4154732/16901230/99772532-4c13-11e6-9a05-2c80acc65784.jpg)

**Comment by rovarma** [comment]
@andregm3 what kind of accessory is that? A custom mounting bracket for the navi to the move? Is it available for purchase somewhere?

**Comment by andregm3** [comment]
http://www.mtbs3d.com/phpBB/viewtopic.php?f=139&t=19998

**Comment by andregm3** [comment]
![4930862644_c78e8b3272_b](https://cloud.githubusercontent.com/assets/4154732/16909583/19b1cd38-4caa-11e6-97b7-006c4498d034.jpg)

**Comment by andregm3** [comment]
![26d27ad8-bb8c-419e-8e07-c771f30c15fb](https://cloud.githubusercontent.com/assets/4154732/16952816/3c0e07dc-4da1-11e6-9113-6fa324d72b30.jpg)

**Comment by cboulay** [comment]
We'll have to work in a way to add an arbitrary transform so that the pose of the PSNavi is the returned pose. @HipsterSloth , do you think it would it be enough to redo the calibration but placing the PSNavi at the calibration points instead of the PSMove?

**Comment by HipsterSloth** [comment]
@cboulay Just placing the attached navi on the calibration points wouldn't be enough to disambiguate between the navi offset and the calibration mat being at a slightly different location. But if we could add a final step at the end of the calibration mat sampling where we have you place the navi on the center and the psmove along the +x axis that would allow us to measure the offset.

**Comment by ReallyDiskun** [comment]
Hi guys, how's the PSNavi support going? I was about to design and 3D-print some kind of rig for the Navi and the Move to go together. Then I realized I haven't heard of Navi support in months  :laughing:

**Comment by gb2111** [comment]
![aaaa](https://cloud.githubusercontent.com/assets/22025896/20100869/9ceaaaee-a5bf-11e6-9bcd-da64f37d7ba2.jpg)

**Comment by ReallyDiskun** [comment]
I can make that indeed. But I was thinking of tilting the navi for a more ergonomic position in any game that involves shooting. Holding the rig like a flashlight gets uncomfortable after a while.

![image](https://cloud.githubusercontent.com/assets/20774380/20101022/5124b75c-a5c0-11e6-9004-3e738fc20425.png)

**Comment by HipsterSloth** [comment]
A mere 8 months later I finally have a test release with PSNavi support in:

https://drive.google.com/open?id=0B5GgUbBXlQ1uTzhQTUJkdmRfeW8

Documentation is here: 

https://github.com/cboulay/PSMoveService/wiki/PSNavi-Setup

Give this a try and let me know how this works.

**Comment by Miomjon** [comment]
Hi, I have tried your experimental build for the PSNavi, and I am close to make it work, but I am facing a small problem. I have the PSNavi connected with a second bluetooth adapter using SCP. Because of this, since the PSNavi is exposed as an Xbox controller, Steam is reading both the XInput directly, as well as the SteamVR input coming from your driver.  Is there a way to 'hide' the XInput to Steam and the games, or have I missed something in your guide (sorry if I did)?

**Comment by HipsterSloth** [comment]
Yeah I ran into that as well. @zelmon64 figured out how to hide the XInput buttons by adjusting the controller mapping settings in BigScreen settings:

https://groups.google.com/d/msg/psmoveservice/KL3b-6X13OQ/OtKKCgfxBAAJ
"The problem in the SteamVR menu is probably because Steam is still recognising the PSNavi as an Xbox controller. I've never been able to find a way to disable this.

[EDIT: I just found that there seems to be a new (well I've never seen it before) way to configure controllers for BPM which treats other controllers like a Steam Controller. Under "Settings -> Base Configurations -> Big Picture configuration" the trigger can be set to "None" so that it doesn't send the conflicting command. After that the trigger works fine in the menu.]"

I need to update my guide with these instructions.
